msal/build.gradle

```diff
@@ -79,6 +89,59 @@ android {
             def fileName = "${archivesBaseName}-${version}.aar"
             outputFileName = fileName
         }
+
+        def buildVariant = variant.name
+        def testTaskName = "test${buildVariant.capitalize()}UnitTest"
+
+        // Tasks for generating Coverage reports(UnitTest & AndroidTests) for every build variant
+        task "${buildVariant}MsalCoverageReport" (type:JacocoReport, dependsOn: ["$testTaskName", "create${buildVariant.capitalize()}CoverageReport"]) {
+            group = "Code Coverage"
+            description = "Generate Msal Coverage Reports(UnitTest & AndroidTests) on the ${buildVariant.capitalize()}"
+
+            def coverageSourceDirs = "${projectDir}/src/main/java"
+            def fileFilter = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test.*']
+            def javaClasses = fileTree(
+                    dir: "${buildDir}/intermediates/javac/${buildVariant}/classes",
+                    excludes: fileFilter
+            )
+            sourceDirectories.from = files([coverageSourceDirs])
+            classDirectories.from = files([javaClasses])
+            executionData.from = fileTree(dir: "$buildDir", includes: [
+                    "jacoco/${testTaskName}.exec"
+            ])
+            reports {
+                xml.enabled = true
+                html.enabled = true
+            }
+        }
+
+        // Tasks for generating Coverage reports(UnitTest) for every build variant
+        task "${buildVariant}MsalUnitTestCoverageReport" (type:JacocoReport, dependsOn: ["$testTaskName"]) {
+            group = "Code Coverage"
+            description = "Generate Msal Coverage Reports(UnitTest) on the ${buildVariant.capitalize()}"
+
+            def coverageSourceDirs = "${projectDir}/src/main/java"
+            def fileFilter = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test.*']
+            def javaClasses = fileTree(
+                    dir: "${buildDir}/intermediates/javac/${buildVariant}/classes",
+                    excludes: fileFilter
+            )
+            sourceDirectories.from = files([coverageSourceDirs])
+            classDirectories.from = files([javaClasses])
+            executionData.from = fileTree(dir: "$buildDir", includes: [
+                    "jacoco/${testTaskName}.exec"
+            ])
+            reports {
+                xml.enabled = true
+                html.enabled = true
+            }
+        }
+
+        // Tasks for generating Coverage reports(AndroidTest) for every build variant
+        task "${buildVariant}MsalAndroidTestCoverageReport" (type:JacocoReport, dependsOn: ["create${buildVariant.capitalize()}CoverageReport"]) {
+            group = "Code Coverage"
+            description = "Generate Msal Coverage Report(AndroidTest) on the ${buildVariant.capitalize()}"
+        }
```

Comment:
Don't we have tasks already (auto generated) for Code Coverage for instrumented tests?

Reply:
Those tasks does not cover all the files for coverage, and misses on showing coverage in file. so added new method and tasks